lib/Sema/TypeChecker.cpp

```diff
@@ -192,17 +192,17 @@ DeclName TypeChecker::getObjectLiteralConstructorName(ObjectLiteralExpr *expr) {
 /// unambiguous name.
 Type TypeChecker::getObjectLiteralParameterType(ObjectLiteralExpr *expr,
                                                 ConstructorDecl *ctor) {
-  Type argType = ctor->getArgumentInterfaceType();
```

Comment:
With some light refactoring in CSDiag, this can be made to speak in terms of params as well.